internal/resource/mutation/mutation.go

```diff
@@ -103,44 +114,47 @@ func apply(path *PathExpr, startIndex int, obj any, value any) error {
 	state := obj
 
 	for i, section := range path.ast.Sections[startIndex:] {
+		isLastSection := startIndex+i == len(path.ast.Sections)-1
+
 		// Map field indexing
-		if section.Field != nil {
+		if section.Field != nil || (section.Index != nil && section.Index.Key != nil) {
```

Comment:
Which scenarios do these 2 conditionals match?

Reply:
`section.Field != nil` is a field like `parent.field` and ` (section.Index != nil && section.Index.Key != nil) ` is a field like `parent["field"]`

Reply:
They were separate branches before with nearly identical code - refactored to flatten them into this branch